### PR TITLE
modify test app

### PR DESCRIPTION
**Pull request by yuqche**
Add auth module test in Unity test app

[AB#66432](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/66432)

### CODE REVIEW

File: Assets/Puppet/PuppetAuth.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    public void SignIn()
+    {
+        var signInTaskTwo = Auth.SignInAsync();
+        signInTaskTwo.ContinueWith(t => 
+        {
+            if (t.Exception != null)
+            {
+                AuthStatus.text = "Sign In failed. User is not authenticated";
```
Comment (guperrot) [inline Assets/Puppet/PuppetAuth.cs]:
```suggestion
                AuthStatus.text = "Sign In failed. User is not authenticated.";
```

File: Assets/Puppet/PuppetAuth.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                ShowLabels(true);
+                AccountIdName.text = "Exception";
+                AccountIdLabel.text = t.Exception.ToString();
+            }
+            else{
+                var userInformation = t.Result;
+                if (userInformation == null)
+                {
+                    AuthStatus.text = "Sign In failed. User is not authenticated";
```
Comment (guperrot) [inline Assets/Puppet/PuppetAuth.cs]:
```suggestion
                    AuthStatus.text = "Sign In failed. User is not authenticated.";
```

File: Assets/Puppet/PuppetAuth.cs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                }
+                else
+                {
+                    AuthStatus.text = "Sign in succeeded. User is authenticated";
```
Comment (guperrot) [inline Assets/Puppet/PuppetAuth.cs]:
```suggestion
                    AuthStatus.text = "Sign in succeeded. User is authenticated.";
```

File: Assets/Puppet/PuppetAuth.cs (hunk 1 above)
Comment (guperrot) [reply Assets/Puppet/PuppetAuth.cs]: Seems like we use this constant twice, might as well define it as a constant.

File: Assets/Puppet/PuppetAuth.cs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        {
+            return null;
+        }
+        string textToDecode = splitedText[1];
+        if (textToDecode.Length % 4 != 0)
+        {
+            int extraChars = 4 - textToDecode.Length % 4;
+            textToDecode = String.Concat(textToDecode, new string('=', extraChars));
+        }
+        byte[] decodedBytes = Convert.FromBase64String (textToDecode);
```
Comment (guperrot) [inline Assets/Puppet/PuppetAuth.cs]: Please reformat the code.

File: Assets/Puppet/PuppetAuth.cs [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        AccessTokenName.enabled = enabled;
+        IdTokenName.enabled = enabled;
+        AccountIdLabel.enabled = enabled;
+        AccessTokenLabel.enabled = enabled;
+        IdTokenLabel.enabled = enabled;
+    }
+
+    public string convertToken(string encodedText)
+    {
+        char[] separator = { '.' };
```
Comment (guperrot) [inline Assets/Puppet/PuppetAuth.cs]: Last time we talked about showing the token in a test app, the decision was just to say "set" or "unset". I don't think we need this.

Comment (yuqche) [reply Assets/Puppet/PuppetAuth.cs]: Remove the function from PuppetAuth.cs. Will include the decode part in the public documentation

File: Assets/Puppet/PuppetAuth.cs [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                    var accountId = userInformation.AccountId;
+                    var accessToken = userInformation.AccessToken;
+                    var idToken = userInformation.IdToken;
+                    AccountIdName.text = "Account Id";
+                    AccessTokenName.text = "Access Token";
+                    IdTokenName.text = "Id Token";
+                    AccountIdLabel.text = accountId;
+                    if (accessToken == null)
+                    {
+                        AccessTokenLabel.text = TokenUnset;
```
Comment (dhei) [inline Assets/Puppet/PuppetAuth.cs]: You may simplify the code into a one-liner `AccessTokenLabel.text = accessToken == null ? TokenUnset : TokenSet;`.

**Review by dhei — APPROVED** [review]

**Pull request merged**